frontend/src/views/SettingWorkspaceMailDelivery.vue

```diff
@@ -0,0 +1,400 @@
+          >
+            <template #menuItem="{ item }">
+              <div class="text-main flex items-center gap-x-2">
+                {{ item }}
+              </div>
+            </template></BBSelect
+          >
+        </div>
+      </div>
+      <!-- Test Send Email To Someone -->
+      <div class="w-full gap-4 mt-8">
+        <div class="min-w-max w-80">
+          <div class="textlabel pl-1">
+            {{ $t("settings.mail-delivery.field.send-test-email-to") }}
+          </div>
+          <BBTextField
+            class="text-main w-full h-max mt-2"
+            :placeholder="'someone@gmail.com'"
+            :value="state.testMailTo"
+            @input="(e: any) => state.testMailTo = e.target.value"
+          />
+        </div>
+      </div>
+      <div class="flex flex-row w-full">
+        <div class="w-auto gap-4 mt-8">
+          <button
+            type="button"
+            class="btn-primary inline-flex justify-center py-2 px-4"
+            :disabled="state.testMailTo === '' || state.isLoading"
+            @click.prevent="testMailDeliverySetting"
+          >
+            {{ $t("settings.mail-delivery.field.send") }}
+          </button>
+          <BBSpin v-if="state.isLoading" class="ml-1" />
+        </div>
+        <div class="w-auto gap-4 mt-8 ml-10">
+          <button
+            type="button"
+            class="btn-primary inline-flex justify-center py-2 px-4"
+            :disabled="!allowMailDeliveryActionButton || state.isLoading"
+            @click.prevent="updateMailDeliverySetting"
+          >
+            {{ mailDeliverySettingButtonText }}
+          </button>
+          <BBSpin v-if="state.isLoading" class="ml-1" />
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+<script lang="ts" setup>
+import { BBCheckbox, BBSelect, BBTextField } from "@/bbkit";
+import { pushNotification } from "@/store";
+import { computed, onMounted, reactive } from "vue";
+import { useI18n } from "vue-i18n";
+import { cloneDeep, isEqual } from "lodash-es";
+import { useWorkspaceMailDeliverySettingStore } from "@/store/modules/workspaceMailDeliverySetting";
+import {
+  SMTPMailDeliverySettingValue,
+  SMTPMailDeliverySettingValue_Authentication,
+  SMTPMailDeliverySettingValue_Encryption,
+} from "@/types/proto/v1/setting_service";
+import { ClientError } from "nice-grpc-web";
+
+interface LocalState {
+  originMailDeliverySetting?: SMTPMailDeliverySettingValue;
+  mailDeliverySetting: SMTPMailDeliverySettingValue;
+  testMailTo: string;
+  isLoading: boolean;
+  useEmptyPassword: boolean;
+}
+const { t } = useI18n();
+
+const state = reactive<LocalState>({
+  mailDeliverySetting: {
+    server: "",
+    port: 587,
+    username: "",
+    password: undefined,
+    from: "",
+    authentication:
+      SMTPMailDeliverySettingValue_Authentication.AUTHENTICATION_PLAIN,
+    encryption: SMTPMailDeliverySettingValue_Encryption.ENCRYPTION_STARTTLS,
+    to: "",
+  },
```

Comment:
nits: extract this part to a utility function sth like `defaultMailDeliverySetting()`.

Reply:
Done.